core/src/banking_stage/unprocessed_transaction_storage.rs

```diff
@@ -1473,6 +1564,44 @@ mod tests {
         std::error::Error,
     };
 
+    #[test]
+    fn transfer_encoding() {
+        let from = Keypair::from_seed(&[1; 32]).unwrap();
+        let to = Pubkey::new_from_array([2; 32]);
+        let amount = 250;
+        let transfer =
+            system_transaction::transfer(&from, &to, amount, Hash::new_from_array([3; 32]));
+
+        let (recovered_to, recovered_amount) = BundleStorage::extract_transfer(
+            &transfer.message.account_keys,
+            &solana_system_program::id(),
+            &transfer.message.instructions[0],
+        )
+        .unwrap();
+        assert_eq!(recovered_to, &to);
+        assert_eq!(recovered_amount, amount);
+    }
+
+    #[test]
+    fn test_priority_sorting_behavior() {
+        let mut priorities = vec![
+            (std::cmp::Reverse(100), 1),
+            (std::cmp::Reverse(200), 2),
+            (std::cmp::Reverse(100), 3),
+            (std::cmp::Reverse(50), 4),
+        ];
+        priorities.sort();
+        assert_eq!(
+            priorities,
+            vec![
+                (std::cmp::Reverse(200), 2),
+                (std::cmp::Reverse(100), 1),
+                (std::cmp::Reverse(100), 3),
+                (std::cmp::Reverse(50), 4),
+            ]
+        );
+    }
```

Comment:
Can we add a second test that sets up 3 bundles (2 with 1 transaction, 1 with 3 transactions). Each tx should have some priority fee and optionally a tip. There should be some other jito tests that create mock bundles that you can use for reference. Without this test its going to be very hard to be confident our priority function works.